scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/package.scala

```diff
@@ -138,5 +139,47 @@ package object avro {
       val param = WriteParam(schema, numShards, suffix, compression)
       self.write(ParquetAvroIO[T](path))(param)
     }
+
+    /**
+     * Save this SCollection of Avro records as a Parquet files written to dynamic destinations.
+     * @param path output location of the write operation
+     * @param schema must be not null if `T` is of type
+     *               [[org.apache.avro.generic.GenericRecord GenericRecord]].
+     * @param numShards number of shards per output directory
+     * @param suffix defaults to .parquet
+     * @param compression defaults to snappy
+     * @param windowFilenameFunction a function which generates a filename for windowed data, whose arguments represent
+     *                                (the shard number,
+     *                                the total number of shards,
+     *                                the bounded window,
+     *                                the pane info for the window)
+     * @param filenameFunction a function which generates a filename, whose arguments represent
+     *                         (the shard number,
+     *                         the total number of shards)
+     */
+    def saveAsDynamicParquetAvroFile(
+      path: String,
+      schema: Schema = WriteParam.DefaultSchema,
+      numShards: Int = WriteParam.DefaultNumShards,
+      suffix: String = WriteParam.DefaultSuffix,
+      compression: CompressionCodecName = WriteParam.DefaultCompression,
+      windowFilenameFunction: (Int, Int, BoundedWindow, PaneInfo) => String =
```

Comment:
Looking at `FilenamePolicy`, these 2 functions are mutually exclusive? Like the write is either windowed if underlying `PCollection` is, or not, but never the same time?

If that's the case, we can use a `Either[WindowFilenameFunction, FilenameFunction]`? Basically requiring the user to always provide 1, but not both.